tests/unit-tests/scene/test_abstract_shell.cpp

```diff
@@ -904,10 +904,42 @@ TEST_F(AbstractShell, does_not_deactivate_parent_when_switching_children)
     msh::FocusController& focus_controller = shell;
     focus_controller.set_focus_to(surface_child_a->session().lock(), surface_child_a);
 
-    EXPECT_CALL(mock_surface, set_focus_state(_)).Times(0);
+    EXPECT_CALL(mock_surface, set_focus_state(mir_window_focus_state_active)).Times(AnyNumber());
     focus_controller.set_focus_to(surface_child_b->session().lock(), surface_child_b);
 }
 
+TEST_F(AbstractShell, makes_parent_active_when_switching_to_child)
+{
+    auto const surface_parent = create_surface(mock_surface);
+
+    NiceMock<mtd::MockSurface> mock_surface_child;
+    auto const surface_child = create_surface(mock_surface_child);
+    ON_CALL(mock_surface_child, parent())
+        .WillByDefault(Return(surface_parent));
+
+    msh::FocusController& focus_controller = shell;
+    focus_controller.set_focus_to(surface_parent->session().lock(), surface_parent);
+
+    EXPECT_CALL(mock_surface, set_focus_state(mir_window_focus_state_active));
+    focus_controller.set_focus_to(surface_child->session().lock(), surface_child);
```

Comment:
This looks like we're testing that the parent surface gets sent a *duplicate* `mir_window_focus_state_active` notification? Am I reading this test correctly?

If so, I don't think that's a useful thing to test; it's testing a quirk of the current implementation rather than a behaviour that we want to preserve.